platform-facet/src/main/java/net/kyori/adventure/platform/facet/Facet.java

```diff
@@ -0,0 +1,635 @@
+          logUnsupported(this, flag);
+        }
+      }
+      return bit;
+    }
+  }
+
+  /**
+   * A facet that listens to boss bar changes, using fake entities.
+   *
+   * @param <V> a viewer type
+   * @param <P> a position type
+   */
+  interface BossBarEntity<V, P> extends BossBar<V>, FakeEntity<V, P> {
+    int OFFSET_PITCH = 30;
+    int OFFSET_YAW = 0;
+    int OFFSET_MAGNITUDE = 40;
+    int INVULNERABLE_KEY = 20;
+    int INVULNERABLE_TICKS = 890;
+
+    @Override
+    default void bossBarPercentChanged(final net.kyori.adventure.bossbar.@NonNull BossBar bar, final float oldPercent, final float newPercent) {
+      this.health(newPercent);
+    }
+
+    @Override
+    default void bossBarNameChanged(final net.kyori.adventure.bossbar.@NonNull BossBar bar, final @NonNull Component oldName, final @NonNull Component newName) {
+      this.name(newName);
+    }
+
+    @Override
+    default void addViewer(final @NonNull V viewer) {
+      this.teleport(viewer, this.createPosition(viewer));
+    }
+
+    @Override
+    default void removeViewer(final @NonNull V viewer) {
+      this.teleport(viewer, null);
+    }
+  }
+
+  /**
+   * A facet for spawning client-side entities.
+   *
+   * @param <V> a viewer type
+   * @param <P> a position type
+   */
+  interface FakeEntity<V, P> extends Position<V, P>, Closeable {
+    /**
+     * Teleports the entity for a viewer.
+     *
+     * @param viewer a viewer
+     * @param position an entity position or {@code null} to remove
+     */
+    void teleport(final @NonNull V viewer, final @Nullable P position);
+
+    /**
+     * Sets the entity metadata.
+     *
+     * @param position a metadata position
+     * @param data a value
+     */
+    void metadata(final int position, final @NonNull Object data);
+
+    /**
+     * Sets the entity visibility.
+     *
+     * @param invisible if invisible
+     */
+    void invisible(final boolean invisible);
+
+    /**
+     * Sets the entity health.
+     *
+     * @param health health level, between 0 and 1
+     */
+    void health(final float health);
+
+    /**
+     * Sets the entity name.
+     *
+     * @param name a name
+     */
+    void name(final @NonNull Component name);
+
+    /**
+     * Remove the entity for all viewers.
+     */
+    @Override
+    void close();
+  }
+
+  Translator TRANSLATOR = new Translator();
+
+  class Translator extends TranslatableComponentRenderer<Locale> {
+    private static final TranslationRegistry TRANSLATIONS = TranslationRegistry.get();
+    private static final Locale DEFAULT_LOCALE = Locale.US; // Default for vanilla clients and servers
+
+    @Override
+    protected @Nullable MessageFormat translate(final @NonNull String key, final @NonNull Locale locale) {
+      final MessageFormat format = TRANSLATIONS.translate(key, locale);
+      if(format != null) return format;
+      if(DEFAULT_LOCALE.equals(locale)) {
+        return new MessageFormat("<missing translation key ''" + key + "''>");
```

Comment:
Can you please return `null` here? Otherwise passing vanilla keys is going to break.

Reply:
11034a62cac1efedb9feba8b037051db46892b6c